buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java

```diff
@@ -94,21 +96,26 @@ public void apply(Project project) {
                 (name, port) -> postProcessFixture.getExtensions()
                     .getByType(ExtraPropertiesExtension.class).set(name, port)
             );
+            extension.fixtures.add(project);
         }
 
-        extension.fixtures.all(fixtureProject -> project.evaluationDependsOn(fixtureProject.getPath()));
+        extension.fixtures
+            .matching(fixtureProject -> fixtureProject.equals(project) == false)
+            .all(fixtureProject ->  project.evaluationDependsOn(fixtureProject.getPath()));
+
+        conditionTaskByType(tasks, extension, getTaskClass("com.carrotsearch.gradle.junit4.RandomizedTestingTask"));
+        conditionTaskByType(tasks, extension, getTaskClass("org.elasticsearch.gradle.test.RestIntegTestTask"));
+        conditionTaskByType(tasks, extension, TestingConventionsTasks.class);
+        conditionTaskByType(tasks, extension, ComposeUp.class);
+
         if (dockerComposeSupported(project) == false) {
             project.getLogger().warn(
                 "Tests for {} require docker-compose at /usr/local/bin/docker-compose or /usr/bin/docker-compose " +
                     "but none could be found so these will be skipped", project.getPath()
             );
-            disableTaskByType(tasks, getTaskClass("com.carrotsearch.gradle.junit4.RandomizedTestingTask"));
```

Comment:
So does this mean we will no longer disable tests when Docker isn't available?

Reply:
We will do  it via  `onlyIf` since `buidFixture` will be auto disabled when there's no docker.

Reply:
And by "auto-disabled" we simply won't add the requisite task dependencies (the early `return`)?

Maybe it'd be clearer to explicitly disable `buildFixture` in the same way if docker isn't available?

Reply:
We do disable `buildDioxture` as in `setEnabled(false)` when docker is not available. 
We do add the task dependencies, it seems sightly less confusing that way since the dependencies will always be the same. We do need to create the task so the configuration phase still runs correctly even without docker

Reply:
I think I'm missing something. The way it's wired up now, if `dockerComposeSupported() == false` then we simply short-circuit the plugin, which results in the test task dependencies never being setup but at no point is the `buildFixture` task explicitly disabled. We just never run the tests, because we never add the dependencies.